File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int n = nums.length;

        ArrayList<Integer> neg = new ArrayList<>();
        ArrayList<Integer> pos = new ArrayList<>();

        int[] res = new int[n];

        for(int num : nums)
        {
            if(num < 0) neg.add(num * num);
            else pos.add(num * num);
        }

        Collections.reverse(neg);

            int i = 0, j = 0;
            int index = 0;
            while(i < pos.size() && j < neg.size())
            {
                if(pos.get(i) <= neg.get(j))
                {
                    res[index++] = pos.get(i);
                    i++;
                }
                else
                {
                    res[index++] = neg.get(j);
                    j++;
                }
            }

            while(i < pos.size())
            {
                res[index++] = pos.get(i);
                i++;   
            }

            while(j < neg.size())
            {
                res[index++] = neg.get(j);
                j++;   
            }
        
        return res;
    }
}

File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        n = len(nums)

        neg = []
        pos = []

        res = []

        for num in nums:
            if num < 0:
                neg.append(num * num)
            else:
                pos.append(num * num)

        neg.reverse()

        i = j = 0

        while(i < len(pos) and j < len(neg)):
            if pos[i] <= neg[j]:
                res.append(pos[i])
                i = i + 1
            else:
                res.append(neg[j])
                j = j + 1
        
        while i < len(pos):
            res.append(pos[i])
            i = i + 1
        
        while j < len(neg):
            res.append(neg[j])
            j = j + 1

        return res
